src/bot.rs

```diff
@@ -200,308 +306,234 @@ impl Bot {
-            db.delete(&key).unwrap();
+    pub fn update_entry(&self, key: &[u8], entry: DbEntry) -> Result<()> {
+        self.db.delete(&key).context(error::Db)?;
+
+        self.db
+            .put(
+                key,
+                serde_json::to_string(&entry)
+                    .context(error::Json)?
+                    .as_bytes(),
+            )
+            .context(error::Db)
+    }
+
+    pub fn act(&mut self) -> Result<()> {
+        for (key, value) in self.db.iterator(IteratorMode::Start) {
+            let entry: DbEntry =
+                serde_json::from_str(&String::from_utf8_lossy(&value)).context(error::Json)?;
+            log::info!("Checking Entry: {:?}", entry);
+
+            match entry {
+                DbEntry::PullRequest(mut data) => {
+                    if data.n_participants >= 2 {
+                        continue;
+                    } else {
+                        data.last_ping = self.ping_person(
+                            &data.engineer,
+                            &data.gh_metadata,
+                            &data.project_room,
+                            data.created_at,
+                            data.last_ping,
+                            "Please assign at least 2 reviewers.",
+                        )?;
+
+                        self.update_entry(&key, DbEntry::PullRequest(data))?;
+                    }
+                }
+
+                DbEntry::ReviewRequest {
+                    created_at,
+                    last_ping,
+                    project_room,
+                    reviewer,
+                    gh_metadata,
+                } => {
+                    let last_ping = self.ping_person(
+                        &reviewer,
+                        &gh_metadata,
+                        &project_room,
+                        created_at,
+                        last_ping,
+                        "Please review",
+                    )?;
+
+                    self.update_entry(
+                        &key,
+                        DbEntry::ReviewRequest {
+                            created_at,
+                            last_ping,
+                            project_room,
+                            reviewer,
+                            gh_metadata,
+                        },
+                    )?;
+                }
+            }
         }
+        Ok(())
     }
 
-    Ok(())
-}
+    pub fn update(&self) -> Result<()> {
+        let mut live_ids = vec![];
+        for repo in self.repositories()? {
+            let prs = self.pull_requests(&repo)?;
+            for pr in prs {
+                let engineer = match self.engineers.get(&pr.user.login) {
+                    Some(eng) => eng,
+                    _ => continue,
+                };
+
+                let id_bytes = pr.id.to_be_bytes();
+                live_ids.push(id_bytes.clone());
+                let np = pr.requested_reviewers.len() as u32;
```

Comment:
What would be the value of `pr.requested_reviewers.len()` if
- 3 reviewers were requested and all added comments/reviews
- no reviewers were requested but 3 people volunteered comments/reviews

Reply:
To answer the first would you be able to comment on this PR? https://github.com/sjeohp-test-org/test-repo/pull/9

Reply:
Commented, although technically I already had since the bot was using my account